src/java/org/apache/cassandra/db/marshal/AbstractCompositeType.java

```diff
@@ -360,7 +360,7 @@ public void checkConstraints(ByteBuffer input, ColumnConstraints constraints) th
         {
             // contraints list should have the exact size of partition key values.
             // Noop constraints are filled for the partition key columns w/o any constraints.
-            throw new IllegalStateException("Partition key values and number of constraints do not match: "
+            throw new IllegalStateException("Partition key value number should be same with number of constraints: "
                                             + partitionKeyValues.size() + "!=" + constraints.getConstraints().size());
```

Comment:
nit: To make the error message a bit more clear, can you indicate the name of the values (size)? The current message prints "1 != 2" for example. How about "The number of constraints (value) should be the same with the number of partition key columns (value)."?